### PR TITLE
LPS-184883 The Edit action in Objects admin interface is labeled as "View" and uses the wrong icon, which might confuse users

### PR DESCRIPTION
**Pull request by ericthmoritsuka**
Related Issue: https://issues.liferay.com/browse/LPS-184883

### CODE REVIEW

**Comment by ericthmoritsuka** [comment]
I used the information in `modules/apps/commerce/commerce-order-rule-web/src/main/java/com/liferay/commerce/order/rule/web/internal/display/context/COREntryDisplayContext.java` to make the alteration. I altered the icon, id, and text, as requested.

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java
Comment (carolmariaabb) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ViewObjectDefinitionsDisplayContext.java:1]: @ericthmoritsuka based on the ticket description, you should also change in the Fields, Relationships, Layouts, Actions, Views, Validations, and State Manager tabs.

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by ericthmoritsuka** [comment]
@carolmariaabb Julia Lira asked me to add a verification to check if the user's role has the necessary permission to edit the field or if he can only view it. I suppose it is something larger and not so simple. I was thinking about adding a condition to check, but then I realized that it will also have to change the way whatever is opened behaves afterward. Any advice on what to do? Just adjusting the icons will be pointless if the main objective is to change the icon and the behavior according to the role.

**Comment by ericthmoritsuka** [comment]
> Hi @ericthmoritsuka,
> 
> > Just adjusting the icons will be pointless if the main objective is to change the icon and the behavior according to the role.
> 
> We are already checking if the user has permission to update a `ObjectField` or a `ObjectAction`, for example. So, in my opinion, you don't need to worry about changing the behavior after clicking view/edit, because if the user doesn't have permission to update the Object Definition, he won't be able to save any changes made by him.
> 
> That said, a possible fix that comes to mind is to do, in the `getFDSActionDropdownItems` of each tab (Fields, Relationships, layouts, Actions, views, Validations and State Manager), something like:
> 
> ```
> FDSActionDropdownItem fdsActionDropdownItem = new FDSActionDropdownItem(
> 	PortletURLBuilder.create(
> 		getPortletURL()
> 	).setMVCRenderCommandName(
> 		....
> 	).setParameter(
> 		....
> 	).setWindowState(
> 		LiferayWindowState.POP_UP
> 	).buildString(),
> 	"view", "view",
> 	LanguageUtil.get(objectRequestHelper.getRequest(), "view"), 
>         ...);
> 
> if (hasUpdateObjectDefinitionPermission()) {
> 	fdsActionDropdownItem.setIcon("pencil");
> 	fdsActionDropdownItem.setLabel(
> 		LanguageUtil.get(objectRequestHelper.getRequest(), "edit"));
> }
> 
> return Arrays.asList(
> 	fdsActionDropdownItem,
>         ....);
> ```
> 
> The `hasUpdateObjectDefinitionPermission` method already exists in the `BaseObjectDefinitionsDisplayContext` class and is responsible for checking if the user has permission to update the Object Definition.
> 
> I'll close this PR for now, but feel free to reach out to the team if you need any assistance with this fix.
> 
> Thanks!

Awesome! I will take a look at it. If I run into any trouble, I'll ask someone. Thanks, @carolmariaabb .

**Pull request closed without merging**